docs/source/en-us/guide/quickstart.md

```diff
@@ -0,0 +1,360 @@
+      {% for item in list %}
+        <div class="item">
+          <a href="{{ item.url }}">{{ item.title }}</a>
+        </div>
+      {% endfor %}
+    </div>
+  </body>
+<html/>
+```
+
+Then add a controller and router.
+
+```js
+// app/controller/news.js
+exports.list = function* newsList() {
+  const dataList = {
+    list: [
+      { id: 1, title: 'this is news 1', url: '/news/1' },
+      { id: 2, title: 'this is news 2', url: '/news/2' }
+    ]
+  };
+  yield this.render('news/list.tpl', dataList);
+};
+
+// app/router.js
+module.exports = app => {
+  app.get('/', 'home');
+  app.get('/news', 'news.list');
+};
+```
+
+Open a browser window and navigate to http://localhost:7001/news.
+You should be able to see the rendered page.
+
+**Tip：In development, Egg enables the [development][egg-development] plug-in by default, which reloads your worker process with when changes are made to your back-end code.**
+
+### Create a service
+
+In practice, controllers usually won't generate data on their own,
+neither will they contain complicated business logic.
+Complicated business logic should instead be abstracted as
+a busineess logic layer, i.e., [service](../basics/service.md).
+
+Let's create a service to fetch data from the
+[HackerNews](https://github.com/HackerNews/API).
+
+```js
+// app/service/news.js
+module.exports = app => {
+  class NewsService extends app.Service {
+    * list(page = 1) {
+      // read config
+      const { serverUrl, pageSize } = this.app.config.new;
+
+      // use build-in http clinet, see https://www.npmjs.com/package/urllib
+      // GET hacker-news api
+      const { data: idList } = yield this.ctx.curl(`${serverUrl}/topstories.json`, {
+        data: {
+          orderBy: '"$key"',
+          startAt: `"${pageSize * (page - 1)}"`,
+          endAt: `"${pageSize * page - 1}"`,
+        },
+        dataType: 'json',
+      });
+
+      // parallel GET detail , see `yield {}` from co
+      const newsList = yield Object.keys(idList).map(key => {
+        const url = `${serverUrl}/item/${idList[key]}.json`;
+        return this.ctx.curl(url, { dataType: 'json' }).then(res => res.data);
+      });
+      return newsList;
+    }
+  }
+  return NewsService;
+};
+```
+
+> Egg has [urllib] built in in order to help you make http requests.
+
+Then slightly modify our previous controller.
+
+```js
+// app/controller/news.js
+exports.list = function* newsList() {
+  const page = this.query.page || 1;
+  const newsList = yield this.service.news.list(page);
+  yield this.render('news/list.tpl', { list: newsList });
+};
+```
+
+### Add Extensions
+
+We might encounter a small problem here.
+The time that we fetched are in Unix Time,
+whereas we want to present them in a more friendly way to read.
+
+Egg provides us with a quick way to extend its functionalities.
+We just need to add extension scripts to the `app/extend` directory.
+For more information, cf. [Extensions](../basics/extend.md).
```

Comment:
cf. |ˌsēˈef|
abbreviation
compare with (used to refer a reader to another written work or another part of the same written work).